File: db-demo/spring-boot-demo/src/main/java/com/example/dbdemo/service/DatabaseInitializer.java
package com.example.dbdemo.service;

import com.example.dbdemo.model.Pet;
import com.example.dbdemo.repository.PetRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

@Component
public class DatabaseInitializer implements CommandLineRunner {

    @Autowired
    private PetRepository petRepository;

    @Override
    public void run(String... args) {
        // Only initialize if database is empty
        if (petRepository.count() == 0) {
            System.out.println("Initializing database with sample pet data...");

            petRepository.save(new Pet("Golden Retriever", "Male", "Max", 5, "Friendly and energetic"));
            petRepository.save(new Pet("Persian Cat", "Female", "Luna", 3, "Calm and loves to cuddle"));
            petRepository.save(new Pet("German Shepherd", "Male", "Rocky", 7, "Loyal and protective"));
            petRepository.save(new Pet("Siamese Cat", "Female", "Bella", 2, "Playful and vocal"));
            petRepository.save(new Pet("Labrador", "Male", "Charlie", 4, "Gentle and loves water"));
            petRepository.save(new Pet("Maine Coon", "Female", "Daisy", 6, "Large and affectionate"));
            petRepository.save(new Pet("Border Collie", "Female", "Molly", 3, "Intelligent and active"));
            petRepository.save(new Pet("Bengal Cat", "Male", "Oliver", 4, "Wild appearance, playful nature"));

            System.out.println("Database initialized with 8 pets");
        } else {
            System.out.println("Database already contains data, skipping initialization");
        }
    }
}

File: db-demo/dotnet-demo/Data/DatabaseInitializer.cs
using DotnetDbDemo.Models;

namespace DotnetDbDemo.Data;

public static class DatabaseInitializer
{
    public static void Initialize(AppDbContext context)
    {
        // Ensure database is created
        context.Database.EnsureCreated();

        // Check if database already has data
        if (context.Pets.Any())
        {
            Console.WriteLine("Database already contains data, skipping initialization");
            return;
        }

        Console.WriteLine("Initializing database with sample pet data...");

        var pets = new Pet[]
        {
            new Pet { Race = "Golden Retriever", Gender = "Male", Name = "Max", Age = 5, Description = "Friendly and energetic" },
            new Pet { Race = "Persian Cat", Gender = "Female", Name = "Luna", Age = 3, Description = "Calm and loves to cuddle" },
            new Pet { Race = "German Shepherd", Gender = "Male", Name = "Rocky", Age = 7, Description = "Loyal and protective" },
            new Pet { Race = "Siamese Cat", Gender = "Female", Name = "Bella", Age = 2, Description = "Playful and vocal" },
            new Pet { Race = "Labrador", Gender = "Male", Name = "Charlie", Age = 4, Description = "Gentle and loves water" },
            new Pet { Race = "Maine Coon", Gender = "Female", Name = "Daisy", Age = 6, Description = "Large and affectionate" },
            new Pet { Race = "Border Collie", Gender = "Female", Name = "Molly", Age = 3, Description = "Intelligent and active" },
            new Pet { Race = "Bengal Cat", Gender = "Male", Name = "Oliver", Age = 4, Description = "Wild appearance, playful nature" }
        };

        context.Pets.AddRange(pets);
        context.SaveChanges();

        Console.WriteLine("Database initialized with 8 pets");
    }
}
